diff -ruN a/src/cmd/go/internal/work/init.go b/src/cmd/go/internal/work/init.go
--- a/src/cmd/go/internal/work/init.go	2018-08-24 21:33:36.000000000 +0200
+++ b/src/cmd/go/internal/work/init.go	2018-11-02 12:07:02.000000000 +0100
@@ -111,7 +111,8 @@
 		ldBuildmode = "c-shared"
 	case "default":
 		switch platform {
-		case "android/arm", "android/arm64", "android/amd64", "android/386":
+		case "linux/amd64", "linux/arm", "linux/arm64", "linux/386", "linux/ppc64le", "linux/s390x",
+			"android/amd64", "android/arm", "android/arm64", "android/386":
 			codegenArg = "-shared"
 			ldBuildmode = "pie"
 		case "darwin/arm", "darwin/arm64":
