diff -ruN a/src/cmd/link/internal/amd64/obj.go b/src/cmd/link/internal/amd64/obj.go
--- a/src/cmd/link/internal/amd64/obj.go	2019-02-26 00:05:57.000000000 +0100
+++ b/src/cmd/link/internal/amd64/obj.go	2019-03-03 16:59:47.000000000 +0100
@@ -62,7 +62,7 @@
 		PEreloc1:         pereloc1,
 		TLSIEtoLE:        tlsIEtoLE,
 
-		Linuxdynld:     "/lib64/ld-linux-x86-64.so.2",
+		Linuxdynld:     "/lib/ld-musl-x86_64.so.1",
 		Freebsddynld:   "/libexec/ld-elf.so.1",
 		Openbsddynld:   "/usr/libexec/ld.so",
 		Netbsddynld:    "/libexec/ld.elf_so",
diff -ruN a/src/cmd/link/internal/arm/obj.go b/src/cmd/link/internal/arm/obj.go
--- a/src/cmd/link/internal/arm/obj.go	2019-02-26 00:05:57.000000000 +0100
+++ b/src/cmd/link/internal/arm/obj.go	2019-03-03 16:59:38.000000000 +0100
@@ -59,7 +59,7 @@
 		Machoreloc1:      machoreloc1,
 		PEreloc1:         pereloc1,
 
-		Linuxdynld:     "/lib/ld-linux.so.3", // 2 for OABI, 3 for EABI
+		Linuxdynld:     "/lib/ld-musl-armhf.so.1",
 		Freebsddynld:   "/usr/libexec/ld-elf.so.1",
 		Openbsddynld:   "/usr/libexec/ld.so",
 		Netbsddynld:    "/libexec/ld.elf_so",
diff -ruN a/src/cmd/link/internal/arm64/obj.go b/src/cmd/link/internal/arm64/obj.go
--- a/src/cmd/link/internal/arm64/obj.go	2019-02-26 00:05:57.000000000 +0100
+++ b/src/cmd/link/internal/arm64/obj.go	2019-03-03 16:59:25.000000000 +0100
@@ -57,7 +57,7 @@
 		Gentext:          gentext,
 		Machoreloc1:      machoreloc1,
 
-		Linuxdynld: "/lib/ld-linux-aarch64.so.1",
+		Linuxdynld: "/lib/ld-musl-aarch64.so.1",
 
 		Freebsddynld:   "XXX",
 		Openbsddynld:   "XXX",
diff -ruN a/src/cmd/link/internal/mips/obj.go b/src/cmd/link/internal/mips/obj.go
--- a/src/cmd/link/internal/mips/obj.go	2019-02-26 00:05:57.000000000 +0100
+++ b/src/cmd/link/internal/mips/obj.go	2019-03-03 16:58:38.000000000 +0100
@@ -60,7 +60,7 @@
 		Gentext:          gentext,
 		Machoreloc1:      machoreloc1,
 
-		Linuxdynld: "/lib/ld.so.1",
+		Linuxdynld: "/lib/ld-musl-mipsle.so.1",
 
 		Freebsddynld:   "XXX",
 		Openbsddynld:   "XXX",
diff -ruN a/src/cmd/link/internal/mips64/obj.go b/src/cmd/link/internal/mips64/obj.go
--- a/src/cmd/link/internal/mips64/obj.go	2019-02-26 00:05:57.000000000 +0100
+++ b/src/cmd/link/internal/mips64/obj.go	2019-03-03 16:58:29.000000000 +0100
@@ -59,7 +59,7 @@
 		Gentext:          gentext,
 		Machoreloc1:      machoreloc1,
 
-		Linuxdynld:     "/lib64/ld64.so.1",
+		Linuxdynld:     "/lib/ld-musl-mips64le.so.1",
 		Freebsddynld:   "XXX",
 		Openbsddynld:   "XXX",
 		Netbsddynld:    "XXX",
diff -ruN a/src/cmd/link/internal/ppc64/obj.go b/src/cmd/link/internal/ppc64/obj.go
--- a/src/cmd/link/internal/ppc64/obj.go	2019-02-26 00:05:57.000000000 +0100
+++ b/src/cmd/link/internal/ppc64/obj.go	2019-03-03 16:58:20.000000000 +0100
@@ -62,7 +62,7 @@
 		Machoreloc1:      machoreloc1,
 
 		// TODO(austin): ABI v1 uses /usr/lib/ld.so.1,
-		Linuxdynld: "/lib64/ld64.so.1",
+		Linuxdynld: "/lib/ld-musl-powerpc64le.so.1",
 
 		Freebsddynld:   "XXX",
 		Openbsddynld:   "XXX",
diff -ruN a/src/cmd/link/internal/s390x/obj.go b/src/cmd/link/internal/s390x/obj.go
--- a/src/cmd/link/internal/s390x/obj.go	2019-02-26 00:05:57.000000000 +0100
+++ b/src/cmd/link/internal/s390x/obj.go	2019-03-03 16:58:10.000000000 +0100
@@ -57,7 +57,7 @@
 		Gentext:          gentext,
 		Machoreloc1:      machoreloc1,
 
-		Linuxdynld: "/lib64/ld64.so.1",
+		Linuxdynld: "/lib/ld-musl-s390x.so.1",
 
 		// not relevant for s390x
 		Freebsddynld:   "XXX",
diff -ruN a/src/cmd/link/internal/x86/obj.go b/src/cmd/link/internal/x86/obj.go
--- a/src/cmd/link/internal/x86/obj.go	2019-02-26 00:05:57.000000000 +0100
+++ b/src/cmd/link/internal/x86/obj.go	2019-03-03 16:57:55.000000000 +0100
@@ -58,7 +58,7 @@
 		Machoreloc1:      machoreloc1,
 		PEreloc1:         pereloc1,
 
-		Linuxdynld:   "/lib/ld-linux.so.2",
+		Linuxdynld:   "/lib/ld-musl-i386.so.1",
 		Freebsddynld: "/usr/libexec/ld-elf.so.1",
 		Openbsddynld: "/usr/libexec/ld.so",
 		Netbsddynld:  "/usr/libexec/ld.elf_so",
